packages/expo-notifications/plugin/src/withNotificationsIOS.ts

```diff
@@ -1,11 +1,67 @@
-import { ConfigPlugin, withEntitlementsPlist } from '@expo/config-plugins';
+import {
+  ConfigPlugin,
+  withEntitlementsPlist,
+  IOSConfig,
+  withXcodeProject,
+} from '@expo/config-plugins';
+import fs from 'fs-extra';
+import path from 'path';
 
-export const withNotificationsIOS: ConfigPlugin<{ mode: 'production' | 'development' }> = (
+import { NotificationsPluginProps } from './withNotifications';
+
+type XcodeProject = any;
+
+export const withNotificationsIOS: ConfigPlugin<NotificationsPluginProps> = (
   config,
-  { mode }
+  { mode = 'development', sounds = [] }
 ) => {
-  return withEntitlementsPlist(config, config => {
+  config = withEntitlementsPlist(config, config => {
     config.modResults['aps-environment'] = mode;
     return config;
   });
+  config = withNotificationSounds(config, { sounds });
+  return config;
+};
+
+export const withNotificationSounds: ConfigPlugin<{ sounds: string[] }> = (config, { sounds }) => {
+  return withXcodeProject(config, config => {
+    setNotificationSounds(sounds, {
```

Comment:
Pass the resolved `config.modRequest.projectName` down since it's a bit safer when the project is in a weird state, this would also help to rely a bit less on `IOSConfig` since that's a massive object that we'll prolly split in the future.